highlighters/main/main-highlighter.zsh

```diff
@@ -751,6 +765,10 @@ _zsh_highlight_main_highlighter_highlight_list()
                            #
                            # The repeat-count word will be handled like a redirection target.
                            this_word=':start::regular:'
+                           ;;
+                        ('function')
+                           next_word=:function:
```

Comment:
Minor: I usually quote `':foo:'` tokens, partly for Vim syntax highlighting purposes.  We should probably be consistent one way or the other.

Reply:
Ah, I was wondering why many were quoted. I'll match.